specs/signing-and-verification-workflow.md

```diff
@@ -110,7 +112,14 @@ The user wants to consume an arbitrary blob only if it was signed by a trusted p
     1. If the signature verification fails, exit.
-1. Calculate the blob's size and verify that it matches the size present in `targetArtifact.size`. Fail signature verification if there is a mismatch.
-1. If provided by the user, verify blob's media type to the one present in `targetArtifact.mediaType`. Fail signature verification if there is a mismatch.
-1. Calculate the digest of the blob using the digest algorithm deduced from signing certificate's public key (see [Algorithm Selection](./signature-specification.md#algorithm-selection)) and match it with the digest specified at `targetArtifact.digest`.  Fail signature verification if there is a mismatch.
-1. If there any user-defined/supplied custom annotations, match them against the ones present in `targetArtifact.annotations`. If they match, signature verification is considered successful.
\ No newline at end of file
+1. **Verify the payload:**
+    1. For COSE format, only [`COSE Hash Envelope`](./signature-envelope-cose.md/#cose-hash-envelope) is supported when verifying arbitrary blobs:
+       1. Deduce payload hash algorithm from signing certificate's public key (see [Algorithm Selection](./signature-specification.md#algorithm-selection)) and match it with the value specified in `protectedHeader.payload-hash-alg (label 258)`. Fail signature verification if there is a mismatch. 
+       1. Calculate the hash of the blob using the hash algorithm from step above, and match it with the value specified at signature envelope `payload`. Fail signature verification if there is a mismatch.
+       1. If provided by the user, verify blob's media type to the one present in `protectedHeader.preimage-content-type (label 259)`. Fail signature verification if there is a mismatch.       
+       1. If there is any user-defined/supplied custom metadata, match them against the ones present in [`Extended attributes`](./signature-specification.md/#extended-attributes). If they match, signature verification is considered successful.
+   1. For JOSE format: 
+      1. Calculate the blob's size and verify that it matches the size present in `targetArtifact.size`. Fail signature verification if there is a mismatch.
+      1. If provided by the user, verify blob's media type to the one present in `targetArtifact.mediaType`. Fail signature verification if there is a mismatch.
+      1. Calculate the digest of the blob using the digest algorithm deduced from signing certificate's public key (see [Algorithm Selection](./signature-specification.md#algorithm-selection)) and match it with the digest specified at `targetArtifact.digest`.  Fail signature verification if there is a mismatch.
+      1. If there is any user-defined/supplied custom annotations, match them against the ones present in `targetArtifact.annotations`. If they match, signature verification is considered successful.
```

Comment:
We still should have a high-level verification protocol without having a switch case for formats.

Reply:
Maybe we could keep it simple by support `COSE Hash Envelope` for blob signature in this release. JOSE format payload could be considered in future releases once there is any request.